piecewise_web/js/ndt.js

```diff
@@ -8,20 +8,17 @@
 
 'use strict';
 
-var WebSocket;
-
-function NDTjs(server, serverPort, serverPath, callbacks, updateInterval, debugLevel) {
+function NDTjs(server, serverPort, serverProtocol, serverPath, callbacks,
+               updateInterval) {
   this.server = server;
   this.serverPort = serverPort || 3001;
   this.serverPath = serverPath || '/ndt_protocol';
+  this.serverProtocol = serverProtocol || 'ws';
```

Comment:
Actually, I'm not sure if we need the conditional here, or for `serverPort`. The way this is currently written, both `serverPort` and `serverProtocol` are required parameters to the function, so it seems the only way they could evaluate to `False` is if the function was explicitly called with those parameters set to something that would evaluate to `False`, which wouldn't make sense really. In summary, I'm wondering if we should just removed all the conditionals here and below, and make all the parameters required?